### PR TITLE
allow user to create timelogs without trello

### PR DESCRIPTION
**Pull request by RomanUkraine**

### CODE REVIEW

File: app/javascript/packs/admin_timelogs/new_timelog.jsx [hunk 1]
```diff
@@ -29,6 +32,7 @@ class NewAdminTimelog extends React.Component {
     Fetch.json(`/api/v1/admin/user/users/${this.state.userId}/timelogs/trello_cards`)
       .then((data) => {
         this.setState({ trelloCards: data });
+        $("#cards_dropdown").prepend("<option value='' selected='selected'></option>");
```
Comment (jizak) [inline app/javascript/packs/admin_timelogs/new_timelog.jsx]: I think it is easier to render HTML using HTML. We don't need to do this with js.

File: app/javascript/packs/admin_timelogs/new_timelog.jsx [hunk 2]
```diff
@@ -65,13 +71,44 @@ class NewAdminTimelog extends React.Component {
     this.setState({ minutes: event.target.value });
   }
 
+  handleTaskChange(event) {
+    this.setState({ task: event.target.value });
+    if (event.target.value.length > 0){
+      $("#cards_dropdown").css("display", "none");
```
Comment (jizak) [inline app/javascript/packs/admin_timelogs/new_timelog.jsx]: can we use `jquery` `.hide` and `.show`?

File: app/javascript/packs/admin_timelogs/new_timelog.jsx [hunk 3]
```diff
@@ -65,13 +71,44 @@ class NewAdminTimelog extends React.Component {
...
+    } else {
+      $("#cards_dropdown").css("display", "block");
+    }
+  }
+
+  handleCardChange(event){
+    this.setState({ card: event.target.value })
+    if (event.target.value != null) {
+      $("#task").css("display", "none");
```
Comment (jizak) [inline app/javascript/packs/admin_timelogs/new_timelog.jsx]: the same is here.

File: app/javascript/packs/admin_timelogs/timelog.jsx [hunk 4]
```diff
@@ -68,26 +75,41 @@ class Timelog extends React.Component {
...
+        (<input
+          type="text"
+          className="form-control"
+          onChange={this.handleTaskChange}
+          defaultValue={this.props.timelog.task}
+        />)
+        :
+        this.props.timelog.task;
+    } else {
+      card_or_task = this.state.editable ?
```
Comment (jizak) [inline app/javascript/packs/admin_timelogs/timelog.jsx]: in js we use js styles for vars.

File: app/javascript/packs/admin_timelogs/timelog.jsx [hunk 5]
```diff
@@ -68,26 +75,41 @@ class Timelog extends React.Component {
...
-      </Select>)
-      :
-      this.props.timelog.trello_card;
+
     const endTime =
       moment(this.props.timelog.end_time).tz('Atlantic/Reykjavik').format('YYYY/MM/DD, HH:mm');
 
+    let card_or_task = null;
+    if (this.state.task != null) {
+      card_or_task = this.state.editable ?
```
Comment (jizak) [inline app/javascript/packs/admin_timelogs/timelog.jsx]: styles.

File: app/javascript/packs/user_timelogs/new_timelog.jsx [hunk 6]
```diff
@@ -63,13 +69,45 @@ class NewUserTimelog extends React.Component {
     this.setState({ minutes: event.target.value });
   }
 
+  handleTaskChange(event) {
+    this.setState({ task: event.target.value });
+    if (event.target.value.length > 0){
+      $("#cards_dropdown").css("display", "none");
```
Comment (jizak) [inline app/javascript/packs/user_timelogs/new_timelog.jsx]: hide. show

File: app/javascript/packs/user_timelogs/new_timelog.jsx [hunk 7]
```diff
@@ -63,13 +69,45 @@ class NewUserTimelog extends React.Component {
...
+    } else {
+      $("#cards_dropdown").css("display", "block");
+    }
+  }
+
+  handleCardChange(event){
+    this.setState({ card: event.target.value })
+    if (event.target.value != null) {
+      $("#task").css("display", "none");
```
Comment (jizak) [inline app/javascript/packs/user_timelogs/new_timelog.jsx]: hide, show

File: app/javascript/packs/user_timelogs/timelog.jsx [hunk 8]
```diff
@@ -68,26 +76,41 @@ class Timelog extends React.Component {
...
-      </Select>)
-      :
-      this.props.timelog.trello_card;
+
     const endTime =
       moment(this.props.timelog.end_time).tz('Atlantic/Reykjavik').format('YYYY/MM/DD, HH:mm');
 
+    let card_or_task = null;
+    if (this.state.task != null) {
+      card_or_task = this.state.editable ?
```
Comment (jizak) [inline app/javascript/packs/user_timelogs/timelog.jsx]: styles

File: app/javascript/packs/user_timelogs/timelog.jsx [hunk 9]
```diff
@@ -68,26 +76,41 @@ class Timelog extends React.Component {
...
+        (<input
+          type="text"
+          className="form-control"
+          onChange={this.handleTaskChange}
+          defaultValue={this.props.timelog.task}
+        />)
+        :
+        this.props.timelog.task;
+    } else {
+      card_or_task = this.state.editable ?
```
Comment (jizak) [inline app/javascript/packs/user_timelogs/timelog.jsx]: styles

**Review by markivancho — APPROVED** [review]

**Pull request closed without merging**